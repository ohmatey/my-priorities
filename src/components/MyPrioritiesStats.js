import React from 'react'
import PropTypes from 'prop-types'
import { compose, withPropsOnChange } from 'recompose'
import { connect } from 'react-redux'

import getTotalPriorityHours from './utils/getTotalPriorityHours'

export const MyPrioritiesStats = ({ totalHours, tallyOfPriorities, timeLeftInDay }) => (
  <div className="box">
    <p>Total hours of priorities: {totalHours}</p>
    <p>Tally of priorities: {tallyOfPriorities}</p>
    <p>Time left in the day: {timeLeftInDay}</p>
  </div>
)

MyPrioritiesStats.propTypes = {
  totalHours: PropTypes.number,
  tallyOfPriorities: PropTypes.number,
  timeLeftInDay: PropTypes.number
}

const mapStateToProps = ({ myPriorities }) => ({
  myPriorities
})

const enhance = compose(
  connect(mapStateToProps),
  withPropsOnChange(
    ['myPriorities'],
    ({ myPriorities }) => {
      const totalHours = getTotalPriorityHours(myPriorities)
      const timeLeftInDay = 24 - totalHours
      const tallyOfPriorities = myPriorities.length

      return {
        totalHours,
        tallyOfPriorities,
        timeLeftInDay
      }
    }
  )
)

export default enhance(MyPrioritiesStats)
