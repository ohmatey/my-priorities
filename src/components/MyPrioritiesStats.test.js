import React from 'react'
import { shallowToJson } from 'enzyme-to-json'

import { MyPrioritiesStats } from './MyPrioritiesStats'

describe('MyPrioritiesList component', () => {
  it('matches snapshot', () => {
    const tree = shallowToJson(
      <MyPrioritiesStats />
    )
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot with props', () => {
    const tree = shallowToJson(
      <MyPrioritiesStats totalHours={10} tallyOfPriorities={2} timeLeftInDay={12} />
    )
    expect(tree).toMatchSnapshot()
  })
})
