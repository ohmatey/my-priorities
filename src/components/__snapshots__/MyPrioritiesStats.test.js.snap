// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPrioritiesList component matches snapshot 1`] = `undefined`;

exports[`MyPrioritiesList component matches snapshot with props 1`] = `undefined`;
